examples/bridge-app/linux/UserInputBackend.cpp

```diff
@@ -0,0 +1,585 @@
+#include <charconv>
+#include <iostream>
+#include <iterator>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace {
+
+std::vector<std::unique_ptr<Device>> g_devices;
+std::vector<std::unique_ptr<DynamicDeviceImpl>> g_device_impls;
+std::unique_ptr<DynamicDeviceImpl> g_pending;
+
+// Pseudo-index representing the device being built.
+static constexpr int kPendingDeviceIndex = -1;
+
+PropagateWriteCB g_write_cb = [](CommonCluster * cluster, chip::EndpointId ep, chip::ClusterId clusterId, chip::AttributeId attr,
+                                 const uint8_t * buffer) -> EmberAfStatus {
+    printf("Write to ep %d cluster %d attribute %d\n", ep, clusterId, attr);
+    cluster->WriteFromBridge(attr, buffer);
+    return EMBER_ZCL_STATUS_SUCCESS;
+};
+
+const char * ParseValue(const std::vector<std::string> & tokens, size_t index, std::string * v)
+{
+    if (index >= tokens.size())
+        return "Missing argument";
+    *v = tokens[index];
+    return nullptr;
+}
+const char * ParseValue(const std::vector<std::string> & tokens, size_t index, std::optional<std::string> * v)
+{
+    if (index < tokens.size())
+        *v = tokens[index];
+    return nullptr;
+}
+
+template <typename T, typename = std::enable_if<std::is_arithmetic<T>::value>>
+const char * ParseValue(const std::vector<std::string> & tokens, size_t index, T * v)
+{
+    if (index >= tokens.size())
+        return "Missing argument";
+    const char * start = tokens[index].data();
+    const char * end   = start + tokens[index].size();
+    auto ret           = std::from_chars(start, end, *v);
+    if (ret.ptr != end)
+        return "Cannot parse as integer";
+    return nullptr;
+}
+
+template <typename T, typename = std::enable_if<std::is_arithmetic<T>::value>>
+const char * ParseValue(const std::vector<std::string> & tokens, size_t index, std::optional<T> * v)
+{
+    uint32_t temp;
+    if (!ParseValue(tokens, index, &temp))
+        *v = temp;
+    return nullptr;
+}
+
+const char * Parse(const std::vector<std::string> & tokens, size_t index)
+{
+    if (index >= tokens.size())
+        return nullptr;
+    return "Unexpected extra tokens";
+}
+
+template <typename T, typename... Args>
+const char * Parse(const std::vector<std::string> & tokens, size_t index, T arg0, Args... args)
+{
+    auto ret = ParseValue(tokens, index, arg0);
+    return ret ? ret : Parse(tokens, index + 1, args...);
+}
+
+void NewDevice(const std::vector<std::string> & tokens)
+{
+    if (g_pending)
+    {
+        printf("Already pending device!\n");
+        return;
+    }
+
+    std::optional<std::string> type;
+    const char * err = Parse(tokens, 0, &type);
+    if (err)
+    {
+        printf("Error: %s\nExpected [prebuilt-device-type]\n", err);
+    }
+    else if (!type)
+    {
+        g_pending = std::make_unique<DynamicDeviceImpl>();
+    }
+    else if (type.value() == "switch")
+    {
+        g_pending = std::make_unique<DynamicSwitchDevice>();
```

Comment:
How is this expected to scale to all device types present in the Matter device type library?